### ISSUE TITLE
`@overload` tags

### ISSUE THREAD

**Issue by castwide** [castwide/solargraph#201]
The `master` branch includes support for the YARD `@overload` tag.

```ruby
class Foo
  # @overload bar(num)
  #   @return [String]
  # @overload bar(num1, num2)
  #   @return [Array<String>]
  def bar *args
  end
end
Foo.new.bar(1)    # <= String
Foo.new.bar(1, 2) # <= Array<String>
```

Internally, Solargraph uses `@overload` for varying return types from `Array#[]`:

```ruby
# @type [Array<String>]
list = ['one', 'two']
list[0..1] # <= Array<String>
list[0, 1] # <= Array<String>
list[0]    # <= String
```

**Comment by minkir014** [comment]
Will this also be published in the release 0.34?

**Comment by castwide** [comment]
Yes.

**Comment by castwide** [comment]
Released in 0.34.0.
